### ISSUE TITLE
translate_alt_id not used in create_go_annots_sprot_trembl.py

### ISSUE THREAD

**Issue by michoug** [lyijin/annotating_proteomes#1]
Hi
Nice pipeline
I was running the `create_go_annots_sprot_trembl.py` script and I encountered the error 

`KeyError: 'GO:0089712 is an alt_id of GO:0070778. Translate it!'`

So I checked the command and saw that TRANSLATE ALT_IDs BEFORE USE was written in the `parse_go_obo.py script`, however, the function that should do that `translate_alt_id` is never called.

Best
Greg

**Comment by lyijin** [comment]
I wrote ``translate_alt_id`` for another script, when I was expecting GO terms to overlap but they didn't--and that was when I discovered that the same GO term can have multiple IDs assigned to them -___-

That said, ``create_go_annots_sprot_trembl.py`` shouldn't need this function to work. I'm a bit mystified. Could you copy and paste the entire crash log from Python 3? I want to see which line triggered the error. Thanks!

**Comment by michoug** [comment]
Here is the error

```
Traceback (most recent call last):
  File "./create_go_annots_sprot_trembl.py", line 82, in <module>
    expanded_go_terms += parse_go_obo.get_all_parent_terms(u)
  File "/data1/refseq/parse_go_obo.py", line 92, in get_all_parent_terms
    alt_to_go_id[query_go_id]+ '. Translate it!')
KeyError: 'GO:0071436 is an alt_id of GO:0036376. Translate it!'
```

Replacing GO:0071436 by GO:0036376 in the text did the trick, but then there was another one, see above

**Comment by lyijin** [comment]
Thanks. Can you modify line 82 to
``expanded_go_terms += parse_go_obo.get_all_parent_terms(parse_go_obo.translate_alt_id(u))``

It will significantly slow down the script (because the translation is an expensive process), but it should suppress the error. Let me know if it works/doesn't work.

Just curious, did you use the pipeline right from the very beginning for your dataset?

**Comment by michoug** [comment]
Yes but in the process of speeding the pipeline I used diamond instead of blastp and also I found out that the `get_top_hit_with_amigo_annot.py` script was super slow (more than 1 night for 10k hits) and rewrote it in Perl (10 min or less), not sure why !

**Comment by michoug** [comment]
The script worked with the correction.
It was actually quite fast

**Comment by lyijin** [comment]
I have to admit that the code in this pipeline is stuff that I've written many many years ago--it is, unfortunately, horribly unoptimised. Speed took a backseat to getting stuff that worked, as I only needed to run the pipeline once per genome (which is... not very often).

If speed is a concern, feel free to fork this and rewrite bits that are slow. Yep, ``get_top_hit_with_amigo_annot.py`` takes ages, i think Python searches through large dictionaries very slowly, perhaps Perl is much better at this use-case.

Hm, if the correction didn't impose a large performance penalty, I'll modify the script to tell others that they can modify it the way I suggested if it crashes at that line for them. Thanks for bringing it to my attention!
